nodejs/aws-infra/experimental/clusterFileSystem.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as mod from ".";
+
+import * as utils from "../utils";
+
+export class ClusterFileSystem extends pulumi.ComponentResource {
+    public readonly instance: aws.efs.FileSystem;
+    public readonly cluster: mod.Cluster;
+    public readonly securityGroup: aws.ec2.SecurityGroup;
+    public readonly mountTargets: aws.efs.MountTarget[];
+    public readonly mountPath: pulumi.Output<string>;
+
+    constructor(name: string,
+                args: ClusterFileSystemArgs,
+                opts: pulumi.CustomResourceOptions = {}) {
+        super("aws-infra:x:ClusterFileSystem", name, {
+            ...args,
+        }, opts);
+
+        const parentOpts = { parent: this };
+
+        const cluster = args.cluster;
+        const instance = new aws.efs.FileSystem(name, args, parentOpts);
+
+        const mountTargets: aws.efs.MountTarget[] = [];
+        const mountPath = utils.ifUndefined(args.mountPath, "/mnt/efs");
+
+        // If requested, add EFS file system and mount targets in each subnet.
+
+        const efsSecurityGroupName = `${name}-fs`;
+        const securityGroup = args.securityGroup || new aws.ec2.SecurityGroup(efsSecurityGroupName, {
+            vpcId: cluster.network.vpcId,
+            ingress: [
+                // Allow NFS traffic from the instance security group
+                {
+                    securityGroups: cluster.instanceSecurityGroups.map(g => g.id),
+                    protocol: "TCP",
+                    fromPort: 2049,
+                    toPort: 2049,
+                },
+            ],
+            tags: { Name: efsSecurityGroupName },
```

Comment:
not sure why we need this tag here since hte name here is the same as the name of the security group we make.